src/extra.ml

```diff
@@ -128,11 +193,47 @@ module Array =
         elements with [eq]. *)
     let equal : 'a eq -> 'a array eq = fun eq a1 a2 ->
       Array.length a1 = Array.length a2 && for_all2 eq a1 a2
+
+    (** [argmax ?cmp a] returns the index of the first maximum of array [a]
+        according to comparison [?cmp].  If [cmp] is not given, defaults to
+        [Pervasives.compare]. *)
+    let argmax : ?cmp:('a -> 'a -> int) -> 'a array -> int =
```

Comment:
I'd rather call the function `max_index`, and an imperative implementation would be better for efficiency here.
```ocaml
let max_index ?cmp:('a -> 'a -> int) -> 'a array -> int =
    fun ?(cmp=Pervasives.compare) arr ->
  let len = Array.length arr in
  if len = 0 then invalid_arg "Extra.Array.max_index";
  let max = ref 0 in
  for i = 1 to len - 1 do
    if cmp arr.(!max) arr.(i) < 0 then max := i
  done; !max
```
Also say in the comment that we return the first occurrence of the maximum element (this may be important in some cases).

(I did not test the function above.)